nano/node/repcrawler.cpp

```diff
@@ -233,16 +235,18 @@ nano::uint128_t nano::rep_crawler::total_weight () const
 void nano::rep_crawler::on_rep_request (std::shared_ptr<nano::transport::channel> channel_a)
 {
 	nano::lock_guard<std::mutex> lock (probable_reps_mutex);
-
-	probably_rep_t::index<tag_channel_ref>::type & channel_ref_index = probable_reps.get<tag_channel_ref> ();
-
-	// Find and update the timestamp on all reps available on the endpoint (a single host may have multiple reps)
-	auto itr_pair = probable_reps.get<tag_channel_ref> ().equal_range (*channel_a);
-	for (; itr_pair.first != itr_pair.second; itr_pair.first++)
+	if (channel_a->get_tcp_endpoint ().address () != boost::asio::ip::address_v6::any ())
 	{
-		channel_ref_index.modify (itr_pair.first, [](nano::representative & value_a) {
-			value_a.last_request = std::chrono::steady_clock::now ();
-		});
+		probably_rep_t::index<tag_channel_ref>::type & channel_ref_index = probable_reps.get<tag_channel_ref> ();
+
+		// Find and update the timestamp on all reps available on the endpoint (a single host may have multiple reps)
+		auto itr_pair = channel_ref_index.equal_range (*channel_a);
+		for (; itr_pair.first != itr_pair.second; itr_pair.first++)
```

Comment:
Optional cleanup for this section, I think this works:
```cpp
// Find and update the timestamp on all reps available on the endpoint (a single host may have multiple reps)
for (auto channel_i : boost::make_iterator_range (probable_reps.get<tag_channel_ref> ().equal_range (*channel_a)))
{
	channel_i.last_request = std::chrono::steady_clock::now ();
}
```

Reply:
I think we can't modify boost multiindex in this way because `last_request` is index

Reply:
Tested this and you're right!